<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate a tidy table with the DE genes from the results of DESeq — deseqresult2DEgenes • ideal</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Generate a tidy table with the DE genes from the results of DESeq — deseqresult2DEgenes"><meta property="og:description" content="Generate a tidy table with the DE genes from the results of DESeq"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ideal</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.25.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/ideal-usersguide.html">ideal User's Guide</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/federicomarini/ideal" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate a tidy table with the DE genes from the results of DESeq</h1>
    <small class="dont-index">Source: <a href="https://github.com/federicomarini/ideal/blob/HEAD/R/res2tbl.R" class="external-link"><code>R/res2tbl.R</code></a></small>
    <div class="hidden name"><code>deseqresult2DEgenes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Generate a tidy table with the DE genes from the results of DESeq</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">deseqresult2DEgenes</span><span class="op">(</span><span class="va">deseqresult</span>, FDR <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>deseqresult</dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/DESeq2/man/DESeqResults.html" class="external-link">DESeqResults</a></code> object</p></dd>


<dt>FDR</dt>
<dd><p>Numeric value, the significance level for thresholding adjusted p-values</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A "tidy" data.frame with only genes marked as differentially expressed</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># with simulated data...</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>package ‘DESeq2’ was built under R version 4.2.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicRanges</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>package ‘GenomicRanges’ was built under R version 4.2.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>package ‘GenomeInfoDb’ was built under R version 4.2.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: SummarizedExperiment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: MatrixGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: matrixStats</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘matrixStats’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:Biobase’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     anyMissing, rowMedians</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘MatrixGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:matrixStats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedSds, rowWeightedVars</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:Biobase’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMedians</span>
<span class="r-in"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/makeExampleDESeqDataSet.html" class="external-link">makeExampleDESeqDataSet</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">100</span>, m <span class="op">=</span> <span class="fl">8</span>, betaSD <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating size factors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">deseqresult2DEgenes</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              id    baseMean log2FoldChange     lfcSE      stat       pvalue</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene70   gene70  614.336021       3.871237 0.3879742  9.978079 1.901121e-23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene45   gene45  222.287630       3.650724 0.4642869  7.863078 3.748065e-15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene76   gene76   85.577500       3.065383 0.4166173  7.357791 1.869782e-13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene43   gene43   28.906129      -4.059469 0.6121258 -6.631756 3.317175e-11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene85   gene85  170.640371       1.959704 0.4025034  4.868789 1.122845e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene65   gene65   20.609754      -3.251515 0.6834003 -4.757849 1.956670e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene100 gene100   26.091350      -2.850521 0.5994365 -4.755334 1.981182e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene46   gene46  104.144159       2.557289 0.5413636  4.723792 2.314869e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4     gene4  105.592293       2.351843 0.5305181  4.433107 9.288471e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene72   gene72   18.759758      -2.774489 0.6258127 -4.433418 9.275101e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene84   gene84  176.019207       2.043265 0.4630786  4.412351 1.022544e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene20   gene20 2045.337892       1.520917 0.3486635  4.362133 1.288005e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene78   gene78   58.768562      -1.865081 0.4704763 -3.964241 7.362991e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene67   gene67  167.422343      -1.590661 0.4036848 -3.940355 8.136098e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene33   gene33   38.648802      -1.802405 0.5039182 -3.576780 3.478522e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene35   gene35  201.599470       1.240847 0.3490444  3.554983 3.780040e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene19   gene19   68.399642       1.984360 0.5644426  3.515610 4.387453e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene34   gene34   37.151210      -1.856673 0.5500423 -3.375509 7.367920e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene87   gene87    5.785544       5.176415 1.6084285  3.218306 1.289501e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene66   gene66    5.209116      -4.501670 1.4264994 -3.155746 1.600881e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene22   gene22   51.738980       1.492834 0.4788053  3.117830 1.821879e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene41   gene41  317.956337       1.147555 0.3924787  2.923866 3.457135e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene80   gene80   30.021956       2.083351 0.7584701  2.746781 6.018339e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene52   gene52  299.798497      -1.117442 0.4123463 -2.709959 6.729154e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene57   gene57   25.922562      -1.467364 0.5696242 -2.576020 9.994477e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene89   gene89   28.001174      -1.344593 0.5354382 -2.511201 1.203213e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene90   gene90  152.459186       1.058588 0.4221603  2.507549 1.215718e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 padj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene70  1.901121e-21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene45  1.874033e-13</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene76  6.232608e-12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene43  8.292938e-10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene85  2.245690e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene65  2.830261e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene100 2.830261e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene46  2.893587e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4   9.288471e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene72  9.288471e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene84  9.295853e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene20  1.073337e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene78  5.663839e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene67  5.811498e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene33  2.319014e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene35  2.362525e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene19  2.580854e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene34  4.093289e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene87  6.786848e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene66  8.004407e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene22  8.675615e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene41  1.571425e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene80  2.616669e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene52  2.803814e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene57  3.997791e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene89  4.502659e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene90  4.502659e-02</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://federicomarini.github.io" class="external-link">Federico Marini</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

