<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Generate a tidy table with the DE genes from the results of DESeq — deseqresult2DEgenes • ideal</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Generate a tidy table with the DE genes from the results of DESeq — deseqresult2DEgenes"><meta property="og:description" content="Generate a tidy table with the DE genes from the results of DESeq"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ideal</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.25.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/ideal-usersguide.html">ideal User's Guide</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/federicomarini/ideal" class="external-link">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate a tidy table with the DE genes from the results of DESeq</h1>
    <small class="dont-index">Source: <a href="https://github.com/federicomarini/ideal/blob/HEAD/R/res2tbl.R" class="external-link"><code>R/res2tbl.R</code></a></small>
    <div class="hidden name"><code>deseqresult2DEgenes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Generate a tidy table with the DE genes from the results of DESeq</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">deseqresult2DEgenes</span><span class="op">(</span><span class="va">deseqresult</span>, FDR <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>deseqresult</dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/DESeq2/man/DESeqResults.html" class="external-link">DESeqResults</a></code> object</p></dd>


<dt>FDR</dt>
<dd><p>Numeric value, the significance level for thresholding adjusted p-values</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A "tidy" data.frame with only genes marked as differentially expressed</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># with simulated data...</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>package ‘DESeq2’ was built under R version 4.2.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicRanges</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>package ‘GenomicRanges’ was built under R version 4.2.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>package ‘GenomeInfoDb’ was built under R version 4.2.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: SummarizedExperiment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: MatrixGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: matrixStats</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘matrixStats’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:Biobase’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     anyMissing, rowMedians</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘MatrixGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:matrixStats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedSds, rowWeightedVars</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:Biobase’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMedians</span>
<span class="r-in"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu">DESeq2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/makeExampleDESeqDataSet.html" class="external-link">makeExampleDESeqDataSet</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">100</span>, m <span class="op">=</span> <span class="fl">8</span>, betaSD <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating size factors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> estimating dispersions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> gene-wise dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> mean-dispersion relationship</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> final dispersion estimates</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> fitting model and testing</span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">deseqresult2DEgenes</span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            id    baseMean log2FoldChange     lfcSE      stat       pvalue</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4   gene4 3120.821725      3.8063896 0.3530980 10.779982 4.279711e-27</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene2   gene2 1037.013760      3.6939738 0.4059346  9.099924 9.039480e-20</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene68 gene68   76.424974      4.8960458 0.5806276  8.432334 3.388391e-17</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene86 gene86   19.822514     -4.1831774 0.7003883 -5.972655 2.334238e-09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene55 gene55  323.579592      1.7094291 0.2897526  5.899617 3.643472e-09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene65 gene65   90.740133      3.1956736 0.5562255  5.745284 9.176690e-09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene71 gene71   16.508670      4.8926836 0.8764034  5.582684 2.368342e-08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene12 gene12   53.772337     -3.0442119 0.5959381 -5.108269 3.251236e-07</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene18 gene18  141.224729      2.0411195 0.4009963  5.090121 3.578356e-07</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene43 gene43   41.884877     -1.9491124 0.4068252 -4.791031 1.659261e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene5   gene5   83.651964      1.8285663 0.3902810  4.685256 2.796097e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene37 gene37  136.228101      1.7706933 0.3852866  4.595782 4.311283e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene11 gene11   23.434808      2.3654455 0.5344758  4.425730 9.611669e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene41 gene41  135.667895      1.8763535 0.4389366  4.274771 1.913338e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene70 gene70   46.370404     -1.8191652 0.4241070 -4.289401 1.791555e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene58 gene58   28.799151     -2.0013248 0.4989021 -4.011458 6.034490e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene67 gene67   23.651935      2.0993548 0.5269827  3.983726 6.784306e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene57 gene57  192.460045      1.4383223 0.3687613  3.900415 9.602783e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene46 gene46   14.228789      3.9260021 1.0354789  3.791484 1.497497e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene30 gene30    7.185623      3.9297742 1.0511930  3.738395 1.851991e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene8   gene8   46.891862     -1.6374226 0.4527326 -3.616754 2.983206e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene36 gene36    9.415856     -2.8173873 0.8383343 -3.360697 7.774619e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene7   gene7   33.285107     -2.1519001 0.6604058 -3.258451 1.120222e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene63 gene63   84.399516      1.4650799 0.4564679  3.209601 1.329191e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene34 gene34  192.969500      1.2632588 0.4028762  3.135601 1.715026e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene85 gene85    5.517670     -4.3845605 1.4835274 -2.955497 3.121660e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene13 gene13   52.304283      1.5996086 0.5540485  2.887127 3.887768e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene61 gene61  751.709859      0.8696914 0.3087156  2.817128 4.845517e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene90 gene90   16.638238     -1.7069060 0.6063020 -2.815274 4.873572e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene47 gene47   11.871055     -1.8862858 0.6977875 -2.703238 6.866753e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene74 gene74   26.830553     -1.6399627 0.6055167 -2.708369 6.761474e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene51 gene51   10.686871     -2.2314023 0.8316309 -2.683164 7.292918e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene88 gene88   23.667852     -1.3338411 0.5151614 -2.589171 9.620722e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene20 gene20    7.756227     -2.0652092 0.8075057 -2.557517 1.054225e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene98 gene98   10.429139     -1.9283676 0.7558285 -2.551330 1.073128e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene94 gene94   40.931973     -1.1840907 0.4732177 -2.502212 1.234202e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene17 gene17   37.716476      1.3242596 0.5395835  2.454226 1.411883e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene40 gene40  103.252471     -0.7499098 0.3114688 -2.407656 1.605530e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                padj</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene4  4.022928e-25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene2  4.248556e-18</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene68 1.061696e-15</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene86 5.485460e-08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene55 6.849728e-08</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene65 1.437681e-07</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene71 3.180345e-07</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene12 3.737394e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene18 3.737394e-06</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene43 1.559705e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene5  2.389392e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene37 3.377172e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene11 6.949976e-05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene41 1.199025e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene70 1.199025e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene58 3.545263e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene67 3.751322e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene57 5.014787e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene46 7.408667e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene30 8.704358e-04</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene8  1.335340e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene36 3.321883e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene7  4.578299e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene63 5.205999e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene34 6.448499e-03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene85 1.128600e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene13 1.353519e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene61 1.579709e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene90 1.579709e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene47 2.082177e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene74 2.082177e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene51 2.142295e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene88 2.740448e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene20 2.882115e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene98 2.882115e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene94 3.222638e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene17 3.586947e-02</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> gene40 3.971574e-02</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="https://federicomarini.github.io" class="external-link">Federico Marini</a>.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

